### PR TITLE
Add the validation function to validate ASM features

### PR DESCRIPTION
**Pull request by nan-yu**
This function can be served as a kpt function to validate if the cluster
configurations meet all ASM requirement.
It can be run as:
```
kpt fn source ${CONFIG_DIR} | kpt fn run --image
gcr.io/kustomize-functions/validate-asm:v0.1.0
```

### CODE REVIEW

**Comment by nan-yu** [comment]
FYI: @justinsb @janetkuo @monopole

File: kustomize-functions/validate-asm/image/main.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+	var errList []error
+
+	meta, err := r.GetMeta()
+	if err != nil {
+		errList = append(errList, err)
+		return errList
+	}
+
+	if meta.Kind == "ContainerCluster" {
```
Comment (janetkuo) [inline kustomize-functions/validate-asm/image/main.go:73]: Make `ContainerCluster` a const. 

Check API version as well?

File: kustomize-functions/validate-asm/image/main.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+			errList = append(errList, err)
+		}
+
+		// validate if mesh_id lable is set
+		if _, err := validateNodeExists(r, meta, "spec", "labels", "mesh_id"); err != nil {
+			errList = append(errList, err)
+		}
+
+		// validate release channel
+		if err := validateNodeValueIn(r, meta, supportedReleaseChannels,"spec", "releaseChannel", "channel"); err != nil {
```
Comment (janetkuo) [inline kustomize-functions/validate-asm/image/main.go:94]: It seems that `spec.releaseChannel` is already a required field.

File: kustomize-functions/validate-asm/Makefile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 The Kubernetes Authors.
```
Comment (janetkuo) [inline kustomize-functions/validate-asm/Makefile:1]: What copyright should we use?

File: kustomize-functions/validate-asm/image/main.go (hunk 1 above)
Comment (nan-yu) [reply kustomize-functions/validate-asm/image/main.go]: DONE

File: kustomize-functions/validate-asm/image/main.go (hunk 2 above)
Comment (nan-yu) [reply kustomize-functions/validate-asm/image/main.go]: yes, if `spec.releaseChannel` node is missing, it also returns an error with a missing node.

File: kustomize-functions/validate-asm/Makefile (hunk 3 above)
Comment (nan-yu) [reply kustomize-functions/validate-asm/Makefile]: Changed to the same Google copyright as the other config files.

**Pull request merged**